File: example/java/config/task/Task2.java
package config.task;

public class Task2 {
    private int taskid;
    private java.util.List<String> name;
    private int nexttask;
    private config.task.completecondition.Completecondition completecondition;
    private int exp;
    private boolean testBool;
    private String testString;
    private config.Position testStruct;
    private java.util.List<Integer> testList;
    private java.util.List<config.Position> testListStruct;
    private java.util.List<config.ai.triggertick.TriggerTick> testListInterface;
    private config.task.Taskextraexp NullableRefTaskid;
    private config.task.Task NullableRefNexttask;

    private Task2() {
    }

    public static Task2 _create(configgen.genjava.ConfigInput input) {
        Task2 self = new Task2();
        self.taskid = input.readInt();
        self.name = new java.util.ArrayList<>();
        for (int c = input.readInt(); c > 0; c--) {
            self.name.add(input.readStr());
        }
        self.nexttask = input.readInt();
        self.completecondition = config.task.completecondition.Completecondition._create(input);
        self.exp = input.readInt();
        self.testBool = input.readBool();
        self.testString = input.readStr();
        self.testStruct = config.Position._create(input);
        self.testList = new java.util.ArrayList<>();
        for (int c = input.readInt(); c > 0; c--) {
            self.testList.add(input.readInt());
        }
        self.testListStruct = new java.util.ArrayList<>();
        for (int c = input.readInt(); c > 0; c--) {
            self.testListStruct.add(config.Position._create(input));
        }
        self.testListInterface = new java.util.ArrayList<>();
        for (int c = input.readInt(); c > 0; c--) {
            self.testListInterface.add(config.ai.triggertick.TriggerTick._create(input));
        }
        return self;
    }

    /**
     * 任务完成条件类型（id的范围为1-100）
     */
    public int getTaskid() {
        return taskid;
    }

    public java.util.List<String> getName() {
        return name;
    }

    public int getNexttask() {
        return nexttask;
    }

    public config.task.completecondition.Completecondition getCompletecondition() {
        return completecondition;
    }

    public int getExp() {
        return exp;
    }

    public boolean getTestBool() {
        return testBool;
    }

    public String getTestString() {
        return testString;
    }

    public config.Position getTestStruct() {
        return testStruct;
    }

    public java.util.List<Integer> getTestList() {
        return testList;
    }

    public java.util.List<config.Position> getTestListStruct() {
        return testListStruct;
    }

    public java.util.List<config.ai.triggertick.TriggerTick> getTestListInterface() {
        return testListInterface;
    }

    public config.task.Taskextraexp nullableRefTaskid() {
        return NullableRefTaskid;
    }

    public config.task.Task nullableRefNexttask() {
        return NullableRefNexttask;
    }

    @Override
    public String toString() {
        return "(" + taskid + "," + name + "," + nexttask + "," + completecondition + "," + exp + "," + testBool + "," + testString + "," + testStruct + "," + testList + "," + testListStruct + "," + testListInterface + ")";
    }

    public void _resolve(config.ConfigMgr mgr) {
        completecondition._resolve(mgr);
        NullableRefTaskid = mgr.task_taskextraexp_All.get(taskid);
        NullableRefNexttask = mgr.task_task_All.get(nexttask);
    }

    public static Task2 get(int taskid) {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.getTaskTask2(taskid);
    }

    public static java.util.Collection<Task2> all() {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.allTaskTask2();
    }

    public static class _ConfigLoader implements config.ConfigLoader {

        @Override
        public void createAll(config.ConfigMgr mgr, configgen.genjava.ConfigInput input) {
            for (int c = input.readInt(); c > 0; c--) {
                Task2 self = Task2._create(input);
                mgr.task_task2_All.put(self.taskid, self);
            }
        }

        @Override
        public void resolveAll(config.ConfigMgr mgr) {
            for (Task2 e : mgr.task_task2_All.values()) {
                e._resolve(mgr);
            }
        }

    }

}

File: example/go/config/Task/Task2.go
package task

type Task2 struct {
    taskid int //任务完成条件类型（id的范围为1-100）
    name []string
    nexttask int
    completecondition config.Task.Completecondition
    exp int
    testBool bool
    testString string
    testStruct config.Position
    testList []int
    testListStruct []config.Position
    testListInterface []config.Ai.TriggerTick
    nullableRefTaskid config_task.Taskextraexp
    nullableRefNexttask config_task.Task
}

//getters
func (t *Task2) GetTaskid() int {
    return t.taskid
}

func (t *Task2) GetName() []string {
    return t.name
}

func (t *Task2) GetNexttask() int {
    return t.nexttask
}

func (t *Task2) GetCompletecondition() config.Task.Completecondition {
    return t.completecondition
}

func (t *Task2) GetExp() int {
    return t.exp
}

func (t *Task2) GetTestBool() bool {
    return t.testBool
}

func (t *Task2) GetTestString() string {
    return t.testString
}

func (t *Task2) GetTestStruct() config.Position {
    return t.testStruct
}

func (t *Task2) GetTestList() []int {
    return t.testList
}

func (t *Task2) GetTestListStruct() []config.Position {
    return t.testListStruct
}

func (t *Task2) GetTestListInterface() []config.Ai.TriggerTick {
    return t.testListInterface
}

//ref properties
func (t *task.task2) GetNullableRefTaskid() config_task.Taskextraexp {
    return t.nullableRefTaskid
}
func (t *task.task2) GetNullableRefNexttask() config_task.Task {
    return t.nullableRefNexttask
}

var all []Task2
func GetAll() []Task2 {
    return all[:len(all)]:len(all)]
}

var allMap map[int]Task2
func Get(key int) (Task2,bool){
    return allMap[key]
}
        public static Task2 Get(int taskid)
        {
            Task2 v;
            return all.TryGetValue(taskid, out v) ? v : null;
        }

        public static List<Task2> All()
        {
            return all.OrderedValues;
        }

        public static List<Task2> Filter(Predicate<Task2> predicate)
        {
            var r = new List<Task2>();
            foreach (var e in all.OrderedValues)
            {
                if (predicate(e))
                    r.Add(e);
            }
            return r;
        }

        internal static void Initialize(Config.Stream os, Config.LoadErrors errors)
        {
            all = new Config.KeyedList<int, Task2>();
            for (var c = os.ReadInt32(); c > 0; c--) {
                var self = _create(os);
                all.Add(self.Taskid, self);
            }
        }

        internal static void Resolve(Config.LoadErrors errors) {
            foreach (var v in All())
                v._resolve(errors);
        }

